feat(platform/github): flag to control whether PRs can be edited by maintainers if `forkToken`is set (#19771)
renovatebot/renovate: docs/usage/configuration-options.md

For example: you're following the `next` tag, but later the stream you actually want is called `stable` instead.
If `next` is no longer getting updates, you must switch your `followTag` to `stable` to get updates again.

## forkModeAllowMaintainerEdits

Use `forkModeAllowMaintainerEdits` to control if maintainers can edit Renovate's pull requests when in fork mode.

> This allows project maintainers to make manual changes to the Renovate PR branch, without needing to create another new PR.

Author: I've tried to add more info here, but it's a little odd because the old workflow is basically *broken* 😄

<!-- prettier-ignore -->
!!! note

>     This option is only relevant if you set `forkToken`.

Author: The `a` is intentionally omitted because it should refer to the `forkToken` option. I think adding `a` would make this a little less clear.

>     This option is only relevant if you set `forkToken`.

Reviewer: It felt a bit wrong to me. 😄 I'll defer to the judgment of a native English speaker on this. 😉

>     This option is only relevant if you set `forkToken`.

Author: me too 😄

For example: you're following the `next` tag, but later the stream you actually want is called `stable` instead.
If `next` is no longer getting updates, you must switch your `followTag` to `stable` to get updates again.

## forkModeDisallowMaintainerEdits

Use `forkModeDisallowMaintainerEdits` to disallow maintainers from editing Renovate's pull requests when in fork mode.

If GitHub pull requests are created from a [fork repository](https://docs.github.com/en/get-started/quickstart/fork-a-repo), the PR author can decide to allow upstream repository to modify the PR directly.

> Allowing maintainers to edit pull requests directly is helpful when Renovate pull requests require additional changes. The reviewer can simply push to the pull request without having to create a new PR. [More details here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).

Reviewer 2: ```suggestion
Allowing maintainers to edit pull requests directly is helpful when Renovate pull requests require additional changes.
The reviewer can simply push to the pull request without having to create a new PR. [More details here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
```
one sentence per line

> You may decide to disallow edits to Renovate pull requests in order to workaround issues in Renovate where modified fork branches are not deleted properly: [See this issue](https://github.com/renovatebot/renovate/issues/16657). If this option is enabled, reviewers will need to create a new PR if additional changes are needed.

Reviewer 2: ```suggestion
You may decide to disallow edits to Renovate pull requests in order to workaround issues in Renovate where modified fork branches are not deleted properly: [See this issue](https://github.com/renovatebot/renovate/issues/16657).
If this option is enabled, reviewers will need to create a new PR if additional changes are needed.
```

For example: you're following the `next` tag, but later the stream you actually want is called `stable` instead.
If `next` is no longer getting updates, you must switch your `followTag` to `stable` to get updates again.

> ## forkModeDisallowMaintainerEdits

Reviewer: ```suggestion
## forkModeBlockMaintainerEdits
```

> Use `forkModeDisallowMaintainerEdits` to disallow maintainers from editing Renovate's pull requests when in fork mode.

Reviewer: ```suggestion
Use `forkModeBlockMaintainerEdits` to block maintainers from editing Renovate's pull requests when in fork mode.
```

If GitHub pull requests are created from a [fork repository](https://docs.github.com/en/get-started/quickstart/fork-a-repo), the PR author can decide to allow upstream repository to modify the PR directly.

Allowing maintainers to edit pull requests directly is helpful when Renovate pull requests require additional changes.
The reviewer can simply push to the pull request without having to create a new PR. [More details here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).

> You may decide to disallow edits to Renovate pull requests in order to workaround issues in Renovate where modified fork branches are not deleted properly: [See this issue](https://github.com/renovatebot/renovate/issues/16657).

Reviewer: ```suggestion
You may decide to block edits to Renovate pull requests in order to workaround issues in Renovate where modified fork branches are not deleted properly: [See this issue](https://github.com/renovatebot/renovate/issues/16657).
```

This sets the level the postUpgradeTask runs on, if set to `update` the postUpgradeTask will be executed for every dependency on the branch.
If set to `branch` the postUpgradeTask is executed for the whole branch.

## prAllowMaintainerEdits

> This setting controls whether pull requests created by Renovate will allow edits by maintainers.

Reviewer: ```suggestion
Use `prAllowMaintainerEdits` to control if maintainers can edit Renovate's pull requests.
```

How about something like this?

Author: thanks!

This sets the level the postUpgradeTask runs on, if set to `update` the postUpgradeTask will be executed for every dependency on the branch.
If set to `branch` the postUpgradeTask is executed for the whole branch.

> ## prAllowMaintainerEdits
> 
> Use `prAllowMaintainerEdits` to control if maintainers can edit Renovate's pull requests.
> 
> When Renovate runs in a fork, this allows project maintainers to make manual changes to the Renovate PR branch, without needing to create another new PR.
> 
> <!-- prettier-ignore -->
> !!! note
>     This option is only relevant if you set `forkToken`.

Reviewer 3: ```suggestion
## forkModeAllowMaintainerEdits

Use `forkModeAllowMaintainerEdits` to control if maintainers can edit Renovate's pull requests when in fork mode.

When Renovate runs in a fork, this allows project maintainers to make manual changes to the Renovate PR branch, without needing to create another new PR.

<!-- prettier-ignore -->
!!! note
    This option is only relevant if you set `forkToken`.
```

This will require reordering within the file too

This sets the level the postUpgradeTask runs on, if set to `update` the postUpgradeTask will be executed for every dependency on the branch.
If set to `branch` the postUpgradeTask is executed for the whole branch.

## prAllowMaintainerEdits

Use `prAllowMaintainerEdits` to control if maintainers can edit Renovate's pull requests.

> When Renovate runs in a fork, this allows project maintainers to make manual changes to the
> Renovate PR branch, without needing to create another new PR.

Reviewer: ```suggestion
When Renovate runs in a fork, this allows project maintainers to make manual changes to the Renovate PR branch, without needing to create another new PR.
```

Author: updated (sorry was OOO until today)
